pkg/istio-agent/agent.go

```diff
@@ -474,26 +476,39 @@ func fileExists(path string) bool {
 }
 
 // Find the root CA to use when connecting to the CA (Istiod or external).
-func (a *Agent) FindRootCAForCA() string {
+func (a *Agent) FindRootCAForCA() (string, error) {
+	var rootCAPath string
+
 	if a.cfg.CARootCerts == security.SystemRootCerts {
-		return ""
+		return "", nil
 	} else if a.cfg.CARootCerts != "" {
-		return a.cfg.CARootCerts
-	} else if a.secOpts.PilotCertProvider == "kubernetes" {
+		rootCAPath = a.cfg.XDSRootCerts
```

Comment:
When a.cfg.CARootCerts != "", why does the PR change the root CA path from a.cfg.CARootCerts to a.cfg.XDSRootCerts?

Reply:
Good catch! Should be CARootCerts, thanks!